### PR TITLE
Add throttle stats to index stats and recovery stats

### PR DESCRIPTION
**Pull request by bleskes**
This commit adds throttling statistics to the index stats API and to the recovery state.

### CODE REVIEW

File: src/main/java/org/elasticsearch/index/recovery/RecoveryStats.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ *
```
Comment (s1monw) [inline src/main/java/org/elasticsearch/index/recovery/RecoveryStats.java]: can we have a quick comment?

File: src/main/java/org/elasticsearch/index/recovery/RecoveryStats.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+    public void incrementRecoveriesAsTarget() {
+        currentAsTarget.incrementAndGet();
+    }
+
+    public void decrementRecoveriesAsTarget() {
+        currentAsTarget.decrementAndGet();
+    }
+
+    public void incrementRecoveriesAsSource() {
```
Comment (s1monw) [inline src/main/java/org/elasticsearch/index/recovery/RecoveryStats.java]: can we fix the naming here somehow? I had a hard time to get what it means maybe something like `incCurrentSourceRecoveries`?

**Comment by s1monw** [comment]
left minor comments

**Comment by bleskes** [comment]
@s1monw I address the comments, improved the test and solved a couple of issue. Can you take another look?

**Comment by bleskes** [comment]
@s1monw ping?

**Comment by s1monw** [comment]
left a minor comment otherwise LGTM

**Pull request closed without merging**